### PR TITLE
AUTH-323: pki: split out konnectivity certs from the rootCA

### PR DESCRIPTION
**Pull request by stlaz**
/assign @csrwng 
/cc @ibihim

### CODE REVIEW

**Comment by stlaz** [comment]
/test e2e-aws
infra seem to have failed

**Comment by stlaz** [comment]
adding root CA in order to make the upgrade test pass

**Comment by stlaz** [comment]
rebased on current master

**Comment by stlaz** [comment]
/test e2e-aws
Odlly enough, the tests are now all failing, while they were green before. Let's see if that's a one-off.

**Comment by stlaz** [comment]
/retest-required
failed on infra

**Comment by stlaz** [comment]
The aws test now only fails on upgrades. I assume that that's because the before-upgrade cluster does not contain the recent fixes

**Comment by stlaz** [comment]
/retest-required
The failure observed MIGHT be an infrastructure issue per @sjenning, let's try to restart the run

**Comment by csrwng** [comment]
@stlaz I tested the control plane upgrade manually and I can see that the konnectivity-agent on the workers cannot authenticate with the konnectivity-server on the control plane side. I'm not sure what the exact reason is, but I see the konnectivity-agent secret tls.crt is still signed by root-ca.

**Comment by stlaz** [comment]
/test e2e-aws
just to see if it's really broken so that I have something to work with next week

**Comment by stlaz** [comment]
/hold cancel
tests passed \o/

File: control-plane-operator/hostedclusterconfigoperator/controllers/resources/konnectivity/reconcile.go [hunk 1]
```diff
@@ -142,9 +150,23 @@ func buildKonnectivityVolumeWorkerAgentCerts(v *corev1.Volume) {
...
+}
+
 func ReconcileKonnectivityAgentSecret(secret, source *corev1.Secret) {
 	secret.Data = map[string][]byte{}
 	for k, v := range source.Data {
 		secret.Data[k] = v
 	}
 }
+
+func CopyConfigMap(cm, source *corev1.ConfigMap) {
```
Comment (csrwng) [inline control-plane-operator/hostedclusterconfigoperator/controllers/resources/konnectivity/reconcile.go:167]: nit, this belongs in https://github.com/openshift/hypershift/tree/main/support/util

**Comment by stlaz** [comment]
/hold cancel
the tests are green again \o/

**Pull request merged**